src/main/java/io/github/fixitlater/quizapi/service/QuestionService.java

```diff
@@ -24,24 +24,10 @@ public QuestionService(QuestionRepository questionRepository) {
         this.questionRepository = questionRepository;
     }
 
-    public QuestionWithAnswersDTO getRandomQuestion() {
-        return QuestionWithAnswersDTO.fromEntity(getRandomQuestionFromList(questionRepository.findAll()));
-    }
-
     public Optional<QuestionWithAnswersDTO> findQuestionById(Long id) {
         return questionRepository.findById(id).map(QuestionWithAnswersDTO::fromEntity);
     }
 
-    public List<QuestionWithAnswersDTO> getAllQuestions() {
-        List<QuestionEntity> questionEntities = questionRepository.findAll();
-        return questionEntities.stream().map(QuestionWithAnswersDTO::fromEntity).collect(Collectors.toList());
-    }
-
-    public List<QuestionWithAnswersDTO> getAllQuestionsFromCategory(String category) {
-        List<QuestionEntity> questionEntities = questionRepository.findByCategory(category);
-        return questionEntities.stream().map(QuestionWithAnswersDTO::fromEntity).collect(Collectors.toList());
-    }
-
     private QuestionEntity getRandomQuestionFromList(List<QuestionEntity> questionEntities) throws NoSuchElementException {
```

Comment:
Moim zdaniem lepiej pozbyć się tego parametru i losować przy pomocy sql - https://stackoverflow.com/questions/580639/how-to-randomly-select-rows-in-sql